src/sgame/components/SpikerComponent.cpp

```diff
@@ -87,44 +65,75 @@ void SpikerComponent::Think(int timeDelta) {
 		float distance = VectorLength(vecToTarget);
 		float effectArea = 2.0f * M_PI * distance * distance; // Half sphere.
 		float targetArea = 0.5f * diameter * diameter;  // Approx. proj. of target on effect area.
-		float expectedDamage = (targetArea / effectArea) * (float)SPIKER_MISSILES *
-		                       (float)BG_Missile(MIS_SPIKER)->damage;
-		float relativeDamage = expectedDamage / durability;
-
-		if (onSameTeam) {
-			friendlyDamage += relativeDamage;
-		} else {
-			if (distance < SPIKER_SENSE_RANGE) {
-				sensing = true;
-				if (sensing && !lastSensing) {
-					RegisterFastThinker();
-				}
+		float damage = (targetArea / effectArea) * (float)SPIKER_MISSILES *
+			(float)BG_Missile(MIS_SPIKER)->damage;
+
+		expectedDamage += damage;
+
+		if (distance < SPIKER_SENSE_RANGE) {
+			sensing = true;
+
+			if (!lastSensing) {
```

Comment:
I assume you wouldn't want this to run more than once per frame, so perhaps the condition should be like `!sensing && !lastSensing`.

Reply:
Thanks, the check was there before but below `sensing = true`, so I removed it assuming it was entirely pointless.